tests/numerics/fv/test_tpsa.py

```diff
@@ -21,1418 +36,1426 @@
 matrices in the dictionary."""
 
 
-def _discretize_get_matrices(grid: pp.Grid, d: dict):
-    """Helper function to discretize with Tpsa and return the dictionary of
-    discretization matrices.
+@pytest.mark.parametrize("g", [pp.CartGrid([2, 2]), pp.CartGrid([2, 2, 2])])
+@pytest.mark.parametrize("driving_bc_type", ["dir", "neu"])
+@pytest.mark.parametrize("tensile", [False, True])
+def test_compression_tension(g: pp.Grid, driving_bc_type: str, tensile: bool):
+    """Assign a compressive or tensile force, check that the resulting displacement is
+    in the correct direction, and that the total pressure is negative.
 
-    Parameters:
-        grid: Grid to discretize.
-        d: Dictionary with parameters.
+    The logic of the test is as follows (assuming a compressive regime, tensile=False,
+    is compressive on the south, east and (if 3d) bottom faces, the remaining faces
+    having neutral conditions. Check that this results in displacement in the negative
+    x-direction and positive y- (and z) direction. The total pressure should be
+    negative, while EK cannot surmise the correct sign of the rotation by physical
```

Comment:
Doesn't the sign of total pressure depend on tensile?

Reply:
Correct. Specified 'negative in compression'.